how-to/hints-and-tips/docs/appassets-and-launch-external-process.md

```diff
@@ -0,0 +1,279 @@
+
+When the index.html page is loaded and requests the launch of the .NET app it already exists on disk and is launched.
+
+```mermaid
+graph TD
+  subgraph PC
+    OpenFin.Application
+    myApp.exe
+    OpenFin.Application -->|1. Connects to Channel API Service| myApp.exe
+    myApp.exe --> |2. Validates and accepts the connection|OpenFin.Application
+    OpenFin.Application <-->|3. Methods can be exposed by client & service | myApp.exe
+  end
+```
+
+Once the .NET app is downloaded as an app asset, launched via launch external process the OpenFin application can connect to a Channel API Service exposed by the .NET application. The .NET application will see the connection request and can validate whether it should accept it. Once connected the OpenFin application can call functions/methods exposed by the .NET app. The .NET app in turn can call functions/methods (if any) exposed by the OpenFin application. The [Channel API](https://developers.openfin.co/of-docs/docs/channels) is a low level API that can be used to build an SDK that exposes a set of functions that can be called by applications.
+
+A basic example of a .NET app exposing a service to a OpenFin application can be seen at the end of this article.
+
+## Common use cases for App Assets and Launch External Process
+
+The general recommendation is to only use app assets and launch external process if you really need to. If you can rely solely on the OpenFin runtime and the fin APIs then you will have the smoothest deployment and upgrade experience. There are some uses cases where this is not possible and here are two of them:
+
+### Native Application Migration
+
+Some companies have native applications that have been used for a number of years. These applications are rich in functionality but the goal is to move to HTML. How do you move to the Web while keeping a native experience in a manageable amount of time?
+
+If your application is portable then it can be packaged and versioned using app assets. With an app asset you can setup an OpenFin application that gets installed and when launched brings down and launches the native application. You can then migrate content to the web in a phased approach and have it loaded into OpenFin. Existing features are still available through the native app that was launched and any data/services that are required can be exposed to the OpenFin Application from the native app using the [Channel API](https://developers.openfin.co/of-docs/docs/channels) (you might decide to migrate the client side features first and expose your backend through your native application until you are ready to also move the backend). You can then chip away and move more and more of the UI to the web until the native app can become headless and kept around for any remaining services that are required. Eventually you may be able to remove the dependency on the app asset and have a pure OpenFin application.
```

Comment:
What does `portable` mean

Reply:
A term used for an app that can be copied and run from disk, usb etc. Maybe a link to the definition? Or just say an app that doesn't require installation

Reply:
Yes a link, or short description in brackets.

Reply:
updated